pkg/signup/signup_service.go

```diff
@@ -213,3 +239,25 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 
 	return signupResponse, nil
 }
+
+func (s *ServiceImpl) GetUserSignup(userID string) (*crtapi.UserSignup, error) {
+	// Retrieve UserSignup resource from the host cluster
+	userSignup, err := s.UserSignups.Get(userID)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return userSignup, nil
+}
+
+func (s *ServiceImpl) UpdateUserSignup(userSignup *crtapi.UserSignup) (*crtapi.UserSignup, error) {
```

Comment:
same here, could we add a comment maybe describing what this does?

Reply:
Comment added.